server/app/jobs/DurableJobRegistry.java

```diff
@@ -0,0 +1,75 @@
+package jobs;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import java.util.HashMap;
+import java.util.Optional;
+
+/**
+ * Provides the means of looking up a {@link DurableJob} by its {@link DurableJobName}. This is
+ * necessary because all {@link DurableJob}s are persisted by {@link models.PersistedDurableJob}
+ * records. Also provides the means of retrieving a list of all recurring jobs with their associated
+ * {@link RecurringJobExecutionTimeResolver}.
+ */
+public final class DurableJobRegistry {
+
+  private final HashMap<String, RegisteredJob> registeredJobs = new HashMap<>();
+
+  @AutoValue
+  public abstract static class RegisteredJob {
```

Comment:
Need a javadoc, include the implication on the optional resolver not being present.

Reply:
Done